### DISCUSSION TITLE
[@types/passport] `AuthenticateOptions.state` is too narrowly defined

### DISCUSSION THREAD

**Discussion by brownoxford** [DefinitelyTyped/DefinitelyTyped#65450] [Issues with a @types package]
The `AuthenticateOptions.state` property is [too narrowly defined](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport/index.d.ts#L173) because it does not allow for `object`.

This causes issues when using strategies like `passport-oauth2`, which allow for custom data to be stored in the state [when it is passed as an object](https://github.com/jaredhanson/passport-oauth2/blob/master/lib/strategy.js#L263-L266).

[a comment by an automated account is not shown]

**Comment by tlaziuk** [comment]
Do you have any suggestion what that type could then be changed to? Or what behavior would you expect?

**Comment by dbnicholson** [comment]
I filed #74200 to address this and added a comment about why I think the type should be `object` and not something narrower.
